.github/workflows/subflow_run_e2e_tests.yaml

```diff
@@ -1,15 +1,45 @@
+name: "E2E Tests"
+
 on:
   workflow_call:
     inputs:
       e2e_filter:
+        type: string
         required: false
+        default: ""
+
+  workflow_dispatch:
+    inputs:
+      e2e_filter:
+        description: "E2E tests filter"
         type: string
+        required: false
+        default: ""
 
 jobs:
   check:
-    name: Run checks
+    name: "Check ${{ matrix.e2e.name }} e2e tests"
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      max-parallel: 4
+      matrix:
+        e2e:
+          - name: "basic"
+            filter: "basic"
+          - name: "selector and remote"
+            filter: "selector || remote"
+          - name: "other"
+            filter: "!basic && !selector && !remote"
+          - name: "${{ inputs.e2e_filter }}"
+            filter: "${{ inputs.e2e_filter }}"
+            from_input: true
+        exclude:
+          - e2e:
+              # run only from_input part when defined and vice versa
+              from_input: ${{ ( ! inputs.e2e_filter ) && true || '' }}
```

Comment:
I think if e2e_filter is provided in workflow_dispatch, all 4 test cases will run, not sure if it was intended